src/App.tsx

```diff
@@ -1,19 +1,68 @@
-import { About } from './components/About/About.tsx';
-import { Footer } from './components/Footer/Footer.tsx';
-import { Header } from './components/Header/Header.tsx';
+import type { FC, MouseEvent } from 'react';
+import { useEffect, useState } from 'react';
 
-import styles from './app.module.css';
+import { About } from '@components/About/About';
+import { Footer } from '@components/Footer/Footer';
+import { Header } from '@components/Header/Header';
+import { ProductsList } from '@components/ProductsList/ProductsList';
+import type { AddToCartHandler } from '@interfaces/Handlers';
+import { PageName } from '@interfaces/PageName';
+import type { Product } from '@interfaces/Product';
+
+import { MOCK_PRODUCTS } from '@/data/mock-products';
+
+import styles from './App.module.css';
+
+type PagesType = {
+    [key in PageName]: React.ReactNode;
+};
+
+const LS_KEY = 'cartMasterAcademy';
+
+const localStorageCart = localStorage.getItem(LS_KEY) ? JSON.parse(localStorage.getItem(LS_KEY)!) : [];
```

Comment:
you make same action here and on lines 29-32.
I think you can keep only those logic in `useEffect` - it works asynchronously and doesn't break the loading of the whole app, like this one.